# Sampling Subset - Crawled Dataset (Beautiful Soup)

Involve just removing Null Values and choosing 5000 rows randomly.

In [49]:
import pandas as pd
import re

In [36]:
dataset = pd.read_csv('/Users/sankeerthana/Documents/NTU/YEAR_4/SEM_2/CZ4034/IR-Sentiment-Analysis/IR-Sentiment-Analysis/skincare_dataset/productReviews_soup.csv')
dataset.head()

Product_ID          Product_Name Reviewer  \
0  6857050914885  Retinol Expert 0.1%   Caressa   
1  6857050914885  Retinol Expert 0.1%     Bryan   
2  6857050914885  Retinol Expert 0.1%   Tristan   
3  6857050914885  Retinol Expert 0.1%    Claire   
4  6857050914885  Retinol Expert 0.1%    Sun M.   

                                Review_Title  \
0                            The Real Deal!    
1                      Gave me a break-out?    
2                                              
3               Lovely and Doesn’t Irritate    
4   It's so great to have pure 0.1% retinol    

                                              Review  
0   I have been using this for 2 weeks now, and I...  
1   I received the sample size of this product an...  
2                                                NaN  
3                                                NaN  
4                                                NaN

### Data Exploration

In [7]:
dataset.shape

(62460, 5)

In [15]:
print("Number of Null Values:")
print(f"Review Title: {dataset['Review_Title'].isna().sum()}")
print(f"Review : {dataset['Review'].isna().sum()}")


Number of Null Values:
Review Title: 1
Review : 5256


The crawled dataset has 62460 rows and 5 columns out of which 5256 rows have null values.

In [41]:
dataset.drop(['Product_Name', 'Reviewer'], axis=1, inplace=True)
dataset

Product_ID                               Review_Title  \
0      6857050914885                            The Real Deal!    
1      6857050914885                      Gave me a break-out?    
2      6857050914885                                              
3      6857050914885               Lovely and Doesn’t Irritate    
4      6857050914885   It's so great to have pure 0.1% retinol    
...              ...                                        ...   
62455  3942309625925                                  Loved it    
62456  3942309625925                                  Soothing    
62457  3942309625925                  Great for sensitive skin    
62458  3942309625925                                       Wow    
62459  3942309625925              Super calming and hydrating!    

                                                  Review  
0       I have been using this for 2 weeks now, and I...  
1       I received the sample size of this product an...  
2                                                    NaN  
3                                                    NaN  
4                                                    NaN  
...                                                  ...  
62455                                          Loved it   
62456   This sheet mask soothes my acne prone skin. M...  
62457   I have very sensitive skin and this was great...  
62458   I got this for its calming effects and wasn't...  
62459   I love this mask! I’m a huge fan of sheet mas...  

[62460 rows x 3 columns]

In [82]:
dataset.dropna(inplace=True)

In [83]:
dataset

Product_ID                    Review_Title  \
0      6857050914885                 The Real Deal!    
1      6857050914885           Gave me a break-out?    
5      6857051275333     The secret to smooth skin!    
6      6857051275333         Love the way it feels!    
7        10245844553                     Refreshing    
...              ...                             ...   
62455  3942309625925                       Loved it    
62456  3942309625925                       Soothing    
62457  3942309625925       Great for sensitive skin    
62458  3942309625925                            Wow    
62459  3942309625925   Super calming and hydrating!    

                                                  Review  
0       I have been using this for 2 weeks now, and I...  
1       I received the sample size of this product an...  
5       This product 100% helps my skin feel smoother...  
6       I use this product as a toner in the mornings...  
7       Cleans off any dead skin but doesn’t leave my...  
...                                                  ...  
62455                                          Loved it   
62456   This sheet mask soothes my acne prone skin. M...  
62457   I have very sensitive skin and this was great...  
62458   I got this for its calming effects and wasn't...  
62459   I love this mask! I’m a huge fan of sheet mas...  

[57201 rows x 3 columns]

In [86]:
print("Number of Null Values:")
print(f"Review Title: {dataset['Review_Title'].isna().sum()}")
print(f"Review : {dataset['Review'].isna().sum()}")

Number of Null Values:
Review Title: 0
Review : 0


In [87]:
sampled_df = dataset.sample(n=5000)
sampled_df

Product_ID                                 Review_Title  \
40281  2074447675461    Fell instantly in love with this product    
60053   263067205641                              Great product!    
24038      348828425   Would Be 5 Stars but They Added Fragrance    
56488  1948686581829      Amazing! Why didn't I buy this sooner?    
16275     2048096901                   I have a combination skin    
...              ...                                          ...   
31606  2032448143429                                       So-so    
28003   263643561993                   Best for deep cystic acne    
60539     3576059269                            Sparkling clean!    
39609  2070553460805                           My face is happy!    
1088     10245844553                          Really good toner!    

                                                  Review  
40281   From the moment I tried this I fell in love i...  
60053   After reading the reviews, I decided to buy t...  
24038   I bought this to replace the old Pure (or wha...  
56488   I love this product! I mix one scoop with my ...  
16275   I have a combination skin and like foam clean...  
...                                                  ...  
31606   After reading raving reviews, I was excited t...  
28003   I have had relatively good success with these...  
60539   Dissolved my make-up and made it feel like my...  
39609   I decided to purchase this since I already ha...  
1088    This toner helped to even out my skin tone. I...  

[5000 rows x 3 columns]

In [89]:
sampled_df.to_csv('/Users/sankeerthana/Documents/NTU/YEAR_4/SEM_2/CZ4034/IR-Sentiment-Analysis/IR-Sentiment-Analysis/skincare_dataset/crawled_reviews_5k.csv')